src/include/smash/kinematics.h

```diff
@@ -211,7 +211,23 @@ inline double plab_from_s(double mandelstam_s, double m_projectile,
 inline double s_from_Ekin(double e_kin, double m_P, double m_T) {
   return m_P * m_P + m_T * m_T + 2 * m_T * (m_P + e_kin);
 }
-
+/** 
+ * Convert E_kin to Mandelstam-s for two beams with total energies and masses 
+ * (E,m) (E_kin gives per nucleon, E=E_kin*A) 
+ *
+ * \param[in] e_kin_p Kinetic energy of projectile [GeV]     
+ * \param[in] e_kin_t Kinetic energy of target [GeV]     
+ * \param[in] m_p     Mass of projectile [GeV]
+ * \param[in] m_t     Mass of target     [GeV]
+ * \return Mandelstam-s [GeV^2]
+ */
+inline double s_from_Ekin(double e_kin_p, double e_kin_t,
+			  double m_p,     double m_t)
+{
+  double pz_p = std::sqrt(e_kin_p * e_kin_p - m_p * m_p);
+  double pz_t = std::sqrt(e_kin_t * e_kin_t - m_t * m_t);
```

Comment:
There also seems to be a different understanding here what e_kin means. In this context not just the kinetic energy is meant, but the total energy of the particle, right?

Reply:
Yes, I believe it is the same as for the "global" `E_Kin` flag

Reply:
Perhaps one should consider even more keys such as 

- `E_Tot` 
- `E_Tot_per_charge` 
- `P_Lab_per_charge` 
- ....

What is most "natural" depends on the user :-)

Reply:
I am pretty sure that E_Kin never means E_Tot in the code, but let me check again. It will probably not make a big difference for your purposes of high energies, but for the lower energies, we are also looking at, it can.

Reply:
First, off, if you're more comfortable making the keyword `E_Tot`, then by all means do so.  What you have in `smash::s_from_Ekin` is the calculation 

    s = m_p^2 + m_t^2 + 2 m_t (m_p+ E_k) = 

where `m_p` and `E_k`  is the mass and kinetic energy of the projectile, respectively, and `m_t` is the mass of the target (at rest).  The total energy of the projectile is  

    E = E_kin + m

(ignoring factors of c), so the above is really 

    s = m_p^2 + m_t^2 + 2 m_t E

For `smash::s_from_plab` you have 

    s = m_p^2 + m_t^2 + 2 m_t sqrt(m^2 + p_z^2) = m_p^2 + m_t^2 + 2 m_t E 

since 

    E = sqrt(m^2 + p_z^2) 

So you are right, for E~m, then it is not the same (my bad, misread what happened in `smash::s_from_Ekin`). If you would like to keep the keyword `E_Kin` (in _addition_ to `E_Tot`), then

    double smash::s_from_ETot(double mp, double mt, double Ep, double Et) {
      double pz_p = std::sqrt(e_kin_p * e_kin_p - m_p * m_p);
      double pz_t = std::sqrt(e_kin_t * e_kin_t - m_t * m_t);
      return std::pow(e_kin_p+e_kin_t,2)-pow(pz_p-pz_t,2);  
    }

    double smash::s_from_EKin(double mp, double mt, double Ep, double Et) {
      return s_from_ETot(mp,mt,Ep+mp,Et+mt);
    }

That should do the trick.